airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -103,6 +104,7 @@ def __init__(
         password: str | None = None,
         key_id: str | None = None,
         secret_id: str | None = None,
+        arn_role: str | None = None,
```

Comment:
@pankajastro - we should rename this to `role_arn` for consistency with the AWS Provider Package.